src/commands/push.js

```diff
@@ -0,0 +1,89 @@
+const { flags } = require('@oclif/command');
+const inquirer = require('inquirer');
+const AbstractAuthenticatedCommand = require('../abstract-authenticated-command');
+const BranchManager = require('../services/branch-manager');
+const ProjectManager = require('../services/project-manager');
+const EnvironmentManager = require('../services/environment-manager');
+const withCurrentProject = require('../services/with-current-project');
+const envConfig = require('../config');
+
+class PushCommand extends AbstractAuthenticatedCommand {
+  static async askForEnvironment(config) {
+    const environemnts = await new EnvironmentManager(config).listEnvironments();
+    const remoteEnvironments = environemnts.filter((environemnt) => environemnt.type === 'remote');
+
+    if (remoteEnvironments.length) {
+      const response = await inquirer.prompt([{
+        name: 'environment',
+        message: 'Select the remote environment you want to push onto',
+        type: 'list',
+        choices: remoteEnvironments.map(
+          (environemnt) => ({ name: environemnt.name, value: environemnt.name }),
+        ),
```

Comment:
1. replace `environemnt` by `environment`
2. We don't need `{ name, value }` since they are identical, we can just use `name` so no need for environment

```suggestion
        choices: remoteEnvironments.map(({ name }) => name),
```